deployment/roles/haystack/files/custom_component.py

```diff
@@ -1,8 +1,36 @@
-from typing import Dict, List
+
+from typing import Dict, List, Optional
+import logging
+
+from haystack.retriever import ElasticsearchRetriever
 from haystack.schema import BaseComponent
 from haystack.retriever.dense import EmbeddingRetriever
 import numpy as np
 from haystack import Document
+logger = logging.getLogger(__name__)
+
+class LabelElasticsearchRetriever(ElasticsearchRetriever):
+
+    def retrieve(self, query: str, filters: dict = None, top_k: Optional[int] = None, index: str = None) -> List[Document]:
+        #  TODO Query labels
+        documents = self.document_store.query(query, None, 1, None, "label_elasticsearch")
+        first_doc = documents[0].to_dict()
+        if len(documents) > 0:
+            logger.warning("retrieving documents from label")
+            documents = self.document_store.get_documents_by_id([first_doc["meta"]["document_id"]], "document_elasticsearch")
+            doc = documents[0].to_dict()
+            doc["meta"]["weight"] = 99
+            logger.warning(doc)
+            doc = Document.from_dict(doc, field_map={})
+            return [doc]
+        else:
+            documents = super().retrieve(query, filters, top_k, "document_elasticsearch")
+            for doc in documents:
+                doc = doc.to_dict()
+                doc["meta"]["weight"] = 1
```

Comment:
same : true, will do

Reply:
I think this `else: ` could be discarded if, in the pipeline, we keep the ESRetriever as it was, and add our LabelESRetriever in parallel to the ESRetriever and the TitleEmbRetriever. I'll do that in a separated commit in case that's not something you want